File: app/widgets/molecular/primers/taxon_auto_complete_field.js
//= require <templates/auto_complete_field>

JooseModule('Molecular.Primers', function () {
  JooseClass('TaxonAutoCompleteField', {
    isa: Templates.AutoCompleteField,
    has: {
      primer: { is: 'ro', required: true, nullable: false },
      object: { is: 'ro', init: function () { return this.primer() } },
      method:        { is: 'ro', init: 'taxon' },
      valueMethod:   { is: 'ro', init: 'rtid' },
      textMethod:    { is: 'ro', init: 'name' },
      width:         { is: 'ro', init: 275 },
      collection: { is: 'ro', init: function () { return (
        Taxon.
          collection({ context: this.context() }).
          select('rtid', 'name')
      ) } }
    }
  })
})

File: app/widgets/molecular/primers/taxon_auto_complete_field.rb
class Molecular::Primers::TaxonAutoCompleteField < Templates::AutoCompleteField
  def initialize options
    @primer ||= options[:primer] || fail('primer not given')
    @model_object ||= @primer
    @attribute_path ||= 'taxon'
    @value_method = 'rtid'
    @text_method = 'name'
    @width = 275
    super
  end
end

